src/utils/runWithTruffle.js

```diff
@@ -1,19 +1,29 @@
-import Web3 from 'web3'
-import TruffleConfig from 'truffle-config'
-import TruffleCompile from 'truffle-workflow-compile'
-import TruffleEnvironment from 'truffle-core/lib/environment'
 import { Logger } from 'zos-lib'
-
 const log = new Logger('RunWithTruffle')
-const options = { logger: console }
-const config = TruffleConfig.detect(options)
+
+export default function runWithTruffle(script, network, compile = false) {
+  try {
+    require('truffle-config').detect({ logger: console })
+  } catch (error) {
+    throw Error('You have to provide a truffle.js file, please remember to initialize your project running "truffle init".')
+  }
+
+  if (compile) compileWithTruffle()
+  initTruffle(network).then(script)
+}
 
 function initTruffle(network) {
   if(!network) throw Error('A network name must be provided to execute the requested action.')
+
+  const TruffleConfig = require('truffle-config')
+  const config = TruffleConfig.detect({ logger: console })
```

Comment:
A minor detail: why not accept the `config` resolved by `TruffleConfig.detect` in L6 here so we don't have to re-walk the file tree looking for the file? Still, it's a minor optimization, we can merge as is :+1: